neo4j_utils/_driver.py

```diff
@@ -1604,9 +1604,9 @@ def last_fail(self):
 
         return self._queries.get('last_failed')
 
-    def get_neo4j_version(self):
+    def get_neo4j_major_version(self) -> int:
```

Comment:
This could be called each time the instance connects to a new server, and accessed as a property, which could be a tuple of integers (who knows if in the future the minor version number will matter too)

Reply:
Now adapted to have a version property